alws/dramatiq/build.py

```diff
@@ -65,6 +73,50 @@ async def _create_log_repo(task_id: int):
         task = await build_node_crud.get_build_task(db, task_id)
         await build_node_crud.create_build_log_repo(db, task)
 
+async def _check_build_and_completed_tasks(
+            db: Session,
+            build_id: int,
+            platform_id: int
+        ) -> bool:
+    build_tasks = (await db.execute(
+        select(func.count()).select_from(models.BuildTask).where(and_(
+            models.BuildTask.build_id == build_id,
+            models.BuildTask.platform_id == platform_id
+        ))
+    )).scalar()
+
+    completed_tasks = (await db.execute(
+        select(func.count()).select_from(models.BuildTask).where(and_(
+            models.BuildTask.build_id == build_id,
+            models.BuildTask.platform_id == platform_id,
+            models.BuildTask.status.in_([
+                BuildTaskStatus.COMPLETED,
+                BuildTaskStatus.FAILED,
+                BuildTaskStatus.EXCLUDED,
+            ])
+        ))
+    )).scalar()
+
+    return completed_tasks==build_tasks
+
+
+async def _all_build_tasks_completed(db: Session, build_task_id: int) -> bool:
+    async with db.begin():
+        build_id, platform_id = (await db.execute(select(
+            models.BuildTask.build_id,
+            models.BuildTask.platform_id,
+        ).where(
+            models.BuildTask.id == build_task_id
+        ))).first()
+
+        all_completed = await _check_build_and_completed_tasks(db, build_id, platform_id)
+        while not all_completed:
```

Comment:
This one will hold function possibly for a very long time. Probably we just need to do one-time check and if not all tasks are completed then just skip creating test tasks